src/components/pages/group/form/speciesCheckBox/index.tsx

```diff
@@ -0,0 +1,167 @@
+import { Box, useDisclosure, Button } from "@chakra-ui/core";
+import { useForm } from "react-hook-form";
+import TextBoxField from "@components/form/text";
+import TextAreaField from "@components/form/textarea";
+import RadioInputField from "@components/form/radio";
+// import * as Yup from "yup";
+import { emit } from "react-gbus";
+import useTranslation from "@configs/i18n/useTranslation";
+import { PageHeading } from "@components/@core/layout";
+import GroupSelector from "./customCheckbox";
+import SelectAsync from "@components/form/select-async";
+import { axQueryTagsByText } from "@services/observation.service";
+import AdminInviteField from "./adminInviteField";
+import MapInputForm from "./MapInputForm";
+import SavingGroup from "../../saving";
+import { SYNC_SINGLE_GROUP } from "@static/events";
+import GroupIconUploader from "../uploader";
+// import Submit from "@components/form/submit-button";
+// import { parseDate } from "@utils/date";
+// import { useStoreState } from "easy-peasy";
+
+function createGroupForm({ speciesGroups, habitats }) {
+  const hform = useForm();
+  const { t } = useTranslation();
+  const { isOpen, onClose } = useDisclosure(true);
+  const adminInviteList = [
+    {
+      name: "founder",
+      textBoxName: "founder-description",
+      fieldGroupTitle: "Invite Founders",
+      isMultiple: true
+    },
+    {
+      name: "moderator",
+      textBoxName: "moderator-description",
+      fieldGroupTitle: "Invite Moderator",
+      isMultiple: true
+    }
+  ];
+
+  const onTagsQuery = async (q) => {
+    const { data } = await axQueryTagsByText(q);
+    return data.map((tag) => ({ label: tag.name, value: tag.id, version: tag.version }));
+  };
+
+  const getAdminUser = (adminList) => {
+    const idsList = [],
+      emailList = [];
+    if (adminList && adminList.length > 0) {
+      adminList.map((item) => {
+        if (item.__isNew__) {
+          emailList.push(item.value);
+        } else {
+          idsList.push(item.value);
+        }
+      });
+    }
+    return { idsList, emailList };
+  };
+  const handleFormSubmit = (e) => {
+    e.preventDefault();
+    const group = hform.getValues();
+    const founder = getAdminUser(group.founder);
+    const moderator = getAdminUser(group.moderator);
+    const payload = {
+      allowUserToJoin: group.group_invitaion,
+      description: group.group_description,
+      icon: group.group_icon,
+      name: group.group_name,
+      neLatitude: group && group.spacial_coverage.ne ? group.spacial_coverage.ne[0] : "",
+      neLongitude: group && group.spacial_coverage.ne ? group.spacial_coverage.ne[1] : "",
+      swLatitude: group && group.spacial_coverage.se ? group.spacial_coverage.se[0] : "",
+      swLongitude: group && group.spacial_coverage.se ? group.spacial_coverage.se[1] : "",
+      languageId: 0,
+      spacialCOverage: group.sGroup ? group.sGroup : [],
+      habitatCoverage: group.habitatCoverage ? group.habitatCoverage : [],
```

Comment:
There's no need to add condition here this could have been `group.habitatCoverage || []` as well

or preferabally name keys appropriately and have a default value so you might have avoided this transformation